### PR TITLE
Builder preparations

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: src/bundle.rs [hunk 1]
```diff
@@ -203,54 +231,48 @@ impl<T: Authorization> Bundle<T> {
...
             flags: self.flags,
             value_balance: self.value_balance,
             anchor: self.anchor,
-            authorization: step(authorization)?,
+            authorization: step(context, authorization)?,
         })
     }
 }
 
-/// Marker for an unauthorized bundle with no proofs or signatures.
```
Comment (nuttycom) [inline src/bundle.rs:246]: Is the idea with this removal that it will be up to the call site to create their own `Unauthorized` type?

**Review by nuttycom — APPROVED** [review]
utACK

(one nonblocking question about usage, since I'm currently making use of the `Unauthorized` type.)

File: src/bundle.rs (hunk 1 above)
Comment (str4d) [reply src/bundle.rs:246]: Yes; the builder PR does this for example (its type includes the proof, and the `bsk` necessary to later compute the binding signature).

Comment (str4d) [reply src/bundle.rs:246]: If it later becomes apparent that we have a commonly-replicated `Unauthorized` type, we can re-introduce this, but I think it was a premature refactor. Only `Authorized` is a known common requirement.

**Comment by ebfull** [comment]
post-merge ACK

**Pull request merged**